frontend/explorer/public/js/hash.js

```diff
@@ -1209,25 +1209,100 @@ function populateHashPage(hash, explorerHash) {
+	var navigation = document.getElementById('nav-links');
+	var transactionSpan = document.createElement('span');
+	navigation.appendChild(document.createTextNode(' > '));
+	navigation.appendChild(transactionSpan);
+	linkHash(transactionSpan, explorerTransaction.id, 'Transaction');
+}
+
+// appendNavigationMenuCoinOuput adds the coin output link to the top navigation menu
+function appendNavigationMenuCoinOutput(explorerHash, hash) {
+	//Coin Ouput
+	if (explorerHash.transactions != null) {
+		for (var i = 0; i < explorerHash.transactions.length; i++) {
+			for (var j = 0; j < explorerHash.transactions[i].coinoutputids.length; j++) {
+				if (explorerHash.transactions[i].coinoutputids[j] == hash) {
+					appendNavigationMenuTransaction(explorerHash.transactions[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.transactions[i].coinoutputids[j], 'Coin Output');
+				}
+			} 
+		}
+	//Coin Ouput - Block Creator Reward
+	} else if (explorerHash.blocks != null) {
+		for (var i = 0; i < explorerHash.blocks.length; i++) {
+			for (var j = 0; j < explorerHash.blocks[i].minerpayoutids.length; j++) {
+				if (explorerHash.blocks[i].minerpayoutids[j] == hash) {
+					appendNavigationMenuBlock(explorerHash.blocks[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.blocks[i].minerpayoutids[j], 'Payout ID');
+				}
+			}
+		}
+	}
+}
+
+// appendNavigationMenuBlockstakeOuput adds the blockstake output link to the top navigation menu
+function appendNavigationMenuBlockstakeOutput(explorerTransactions, hash) {
+	for (var i = 0; i < explorerTransactions.length; i++) {
+		for (var j = 0; j < explorerTransactions[i].blockstakeoutputids.length; j++) {
+			if (explorerTransactions[i].blockstakeoutputids[j] == hash) {
+				appendNavigationMenuTransaction(explorerTransactions[i]);
+				var navigation = document.getElementById('nav-links');
+				var outputSpan = document.createElement('span');
+				navigation.appendChild(document.createTextNode(' > '));
+				navigation.appendChild(outputSpan);
+				linkHash(outputSpan, explorerTransactions[i].blockstakeoutputids[j], 'Blockstake Output');
+			}
+		}
+	}
+}
+
+// appendNavigationMenuUnlockHash adds the unlock hash link to the top navigation menu
+function appendNavigationMenuUnlockHash(hash) {
+	var navigation = document.getElementById('nav-links');
+	var unlockSpan = document.createElement('span');
+	navigation.appendChild(unlockSpan);
+	switch(hash.substring(0,2)) {
+		case "01": linkHash(unlockSpan, hash, 'Wallet'); break;
+		case "02": linkHash(unlockSpan, hash, 'Multisig Wallet'); break;
```

Comment:
`02` is an Atomic Swap Contract, and `03` is the Multisig Wallet. Don't forget about `"00"` by the way, the Free-for-all Wallet.

Reply:
Added Free-for-all Wallet
Swapped Multisig Wallet and Atomic Swap Contract